src/login/logind-user.c

```diff
@@ -523,7 +525,11 @@ int user_stop(User *u, bool force) {
                         r = k;
         }
 
-        user_stop_service(u, force);
+        r = manager_unit_is_active(u->manager, u->service, &error);
+        if (r < 0)
+                log_debug_errno(r, "Failed to determine whether unit '%s' is active, ignoring: %s", u->service, bus_error_message(&error, r));
+        if (r != 0)
+                user_stop_service(u, force);
```

Comment:
Hmm, is this chunk needed at all? `manager_stop_unit()` calls `StopUnit`, which should be a noop if the unit is not running.

Reply:
Actually, StopUnit() on non-existing unit returns `org.freedesktop.systemd1.NoSuchUnit` and I think we want to avoid that.

Reply:
I still think it'd be better to call `user_stop_service` unconditionally, and check if `org.freedesktop.systemd1.NoSuchUnit` was returned and treat that as success.